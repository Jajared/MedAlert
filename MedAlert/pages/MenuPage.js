import { Text, SafeAreaView, StyleSheet, View, TouchableOpacity, Image, Alert, StatusBar } from "react-native";
import BackNavBar from "../components/BackNavBar/BackNavBar";
import { MaterialCommunityIcons, MaterialIcons, AntDesign, Ionicons, Feather } from "@expo/vector-icons";

export default function ProfilePage({ navigation, userInformation, resetScheduledItems, setIsNotificationReset, onSignOut }) {
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <BackNavBar navigation={navigation} title="Menu" />
      <View style={styles.mainSettingsSection}>
        <TouchableOpacity style={styles.settingsItem} onPress={() => navigation.navigate("Update Account")}>
          <MaterialCommunityIcons name="account" size={30} color="black" style={{ flex: 1 }} />
          <View style={styles.settingInfo}>
            <Text style={styles.settingsTitle}>Update Account</Text>
            <Text style={styles.settingDescription}>Make changes to your account</Text>
          </View>
          <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.settingsItem}
          onPress={() => {
            resetScheduledItems();
            setIsNotificationReset(true);
          }}
        >
          <MaterialIcons name="supervisor-account" size={30} color="black" style={{ flex: 1 }} />
          <View style={styles.settingInfo}>
            <Text style={styles.settingsTitle}>Saved Guardian</Text>
            <Text style={styles.settingsDescription}>Manage your guardian information</Text>
          </View>
          <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
        </TouchableOpacity>
        <TouchableOpacity style={styles.settingsItem} onPress={() => navigation.navigate("Medication Database")}>
          <AntDesign name="database" size={30} color="black" style={{ flex: 1 }} />
          <View style={styles.settingInfo}>
            <Text style={styles.settingsTitle}>Medication Database</Text>
            <Text style={styles.settingsDescription}>Search through our database for your medication</Text>
          </View>
          <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.settingsItem}
          onPress={() => {
            onSignOut().then(() => {
              navigation.navigate("Login");
            });
          }}
        >
          <Ionicons name="exit-outline" size={30} color="black" style={{ flex: 1 }} />
          <View style={styles.settingInfo}>
            <Text style={styles.settingsTitle}>Log Out</Text>
            <Text style={styles.settingsDescription}>Log out of your existing account</Text>
          </View>
          <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
        </TouchableOpacity>
      </View>
      <View style={styles.moreSettingsSection}>
        <Text style={{ fontSize: 15, fontWeight: "bold", flex: 1 }}>More</Text>
        <View style={styles.moreSettingsContainer}>
          <TouchableOpacity style={styles.settingsItem}>
            <MaterialIcons name="support-agent" size={24} color="black" style={{ flex: 1 }} />
            <View style={styles.settingInfo}>
              <Text style={styles.settingsTitle}>Help & Support</Text>
            </View>
            <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
          </TouchableOpacity>
          <TouchableOpacity style={styles.settingsItem}>
            <Ionicons name="ios-heart-circle" size={24} color="black" style={{ flex: 1 }} />
            <View style={styles.settingInfo}>
              <Text style={styles.settingsTitle}>About App</Text>
            </View>
            <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
          </TouchableOpacity>
        </View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    alignItems: "center",
    backgroundColor: "#fff",
    flex: 1,
    width: "100%",
  },
  mainSettingsSection: {
    flex: 5,
    flexDirection: "column",
    borderColor: "black",
    borderWidth: 1,
    margin: 10,
    width: "90%",
    borderRadius: 15,
  },
  settingsItem: {
    flex: 1,
    padding: 10,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  settingInfo: {
    flex: 6,
  },
  settingsTitle: {
    fontSize: 16,
    fontWeight: "bold",
  },
  settingDescription: {
    fontSize: 12,
  },
  moreSettingsSection: {
    flex: 3,
    flexDirection: "column",
    margin: 10,
    width: "90%",
  },
  moreSettingsContainer: {
    flex: 4,
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 15,
  },
});
