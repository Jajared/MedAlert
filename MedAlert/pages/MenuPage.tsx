import { Text, SafeAreaView, StyleSheet, View, TouchableOpacity, StatusBar } from "react-native";
import BackNavBar from "../components/BackNavBar";
import { MaterialCommunityIcons, MaterialIcons, AntDesign, Ionicons, Feather } from "@expo/vector-icons";

export default function ProfilePage({ navigation, onSignOut }) {
  return (
    <SafeAreaView style={styles.container} testID="MenuPage">
      <StatusBar barStyle="dark-content" />
      <BackNavBar navigation={navigation} title="Menu" />
      <View style={styles.mainSettingsSection}>
        <TouchableOpacity style={styles.settingsItem} onPress={() => navigation.navigate("Update Account")} testID="UpdateAccountButton">
          <MaterialCommunityIcons name="account" size={30} color="black" style={{ flex: 1 }} />
          <View style={styles.settingInfo}>
            <Text style={styles.settingsTitle}>Update Account</Text>
            <Text style={styles.settingDescription}>Make changes to your account</Text>
          </View>
          <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
        </TouchableOpacity>
        <TouchableOpacity style={styles.settingsItem} onPress={() => navigation.navigate("Guardian Home")} testID="GuardianButton">
          <MaterialIcons name="supervisor-account" size={30} color="black" style={{ flex: 1 }} />
          <View style={styles.settingInfo}>
            <Text style={styles.settingsTitle}>Saved Guardian</Text>
            <Text style={styles.settingDescription}>Manage your guardian information</Text>
          </View>
          <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
        </TouchableOpacity>
        <TouchableOpacity style={styles.settingsItem} onPress={() => navigation.navigate("Medication Database")} testID="MedicationDatabaseButton">
          <AntDesign name="database" size={30} color="black" style={{ flex: 1 }} />
          <View style={styles.settingInfo}>
            <Text style={styles.settingsTitle}>Medication Database</Text>
            <Text style={styles.settingDescription}>Search through our database for your medication</Text>
          </View>
          <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
        </TouchableOpacity>
        <TouchableOpacity
          style={styles.settingsItem}
          onPress={async () => {
            await onSignOut();
            navigation.navigate("Login");
          }}
          testID="LogoutButton"
        >
          <Ionicons name="exit-outline" size={30} color="black" style={{ flex: 1 }} />
          <View style={styles.settingInfo}>
            <Text style={styles.settingsTitle}>Log Out</Text>
            <Text style={styles.settingDescription}>Log out of your existing account</Text>
          </View>
          <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
        </TouchableOpacity>
      </View>
      <View style={styles.moreSettingsSection}>
        <Text style={{ fontSize: 15, fontWeight: "bold", flex: 1 }}>More</Text>
        <View style={styles.moreSettingsContainer}>
          <TouchableOpacity style={styles.settingsItem} onPress={() => navigation.navigate("Help")} testID="HelpButton">
            <MaterialIcons name="support-agent" size={24} color="black" style={{ flex: 1 }} />
            <View style={styles.settingInfo}>
              <Text style={styles.settingsTitle}>Help & Support</Text>
            </View>
            <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
          </TouchableOpacity>
          <TouchableOpacity style={styles.settingsItem} onPress={() => navigation.navigate("About")} testID="AboutButton">
            <Ionicons name="ios-heart-circle" size={24} color="black" style={{ flex: 1 }} />
            <View style={styles.settingInfo}>
              <Text style={styles.settingsTitle}>About App</Text>
            </View>
            <Feather name="arrow-right" size={24} color="black" style={{ flex: 1 }} />
          </TouchableOpacity>
        </View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    alignItems: "center",
    backgroundColor: "#fff",
    flex: 1,
    width: "100%",
  },
  mainSettingsSection: {
    flex: 5,
    flexDirection: "column",
    borderColor: "black",
    borderWidth: 1,
    margin: 10,
    width: "90%",
    borderRadius: 15,
  },
  settingsItem: {
    flex: 1,
    padding: 10,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
  },
  settingInfo: {
    flex: 6,
  },
  settingsTitle: {
    fontSize: 16,
    fontWeight: "bold",
  },
  settingDescription: {
    fontSize: 12,
  },
  moreSettingsSection: {
    flex: 3,
    flexDirection: "column",
    margin: 10,
    width: "90%",
  },
  moreSettingsContainer: {
    flex: 4,
    borderColor: "black",
    borderWidth: 1,
    borderRadius: 15,
  },
});
