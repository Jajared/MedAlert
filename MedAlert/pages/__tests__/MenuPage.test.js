import React from "react";
import renderer from "react-test-renderer";

import MenuPage from "../MenuPage";

describe("<MenuPage/>", () => {
  it("has 3 child", () => {
    const tree = renderer.create(<MenuPage />).toJSON();
    expect(tree.children.length).toBe(3);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<MenuPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
