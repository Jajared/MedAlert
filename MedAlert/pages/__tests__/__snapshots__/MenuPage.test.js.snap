// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuPage/> renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "width": "90%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": -20,
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginLeft": 16,
        }
      }
    >
      Menu
    </Text>
  </View>
  <View
    style={
      {
        "borderColor": "black",
        "borderRadius": 15,
        "borderWidth": 1,
        "flex": 5,
        "flexDirection": "column",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 30,
            },
            {
              "flex": 1,
            },
            {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        󰀄
      </Text>
      <View
        style={
          {
            "flex": 6,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Update Account
        </Text>
        <Text
          style={
            {
              "fontSize": 12,
            }
          }
        >
          Make changes to your account
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 24,
            },
            {
              "flex": 1,
            },
            {
              "fontFamily": "feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 30,
            },
            {
              "flex": 1,
            },
            {
              "fontFamily": "material",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          {
            "flex": 6,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Saved Guardian
        </Text>
        <Text
          style={
            {
              "fontSize": 12,
            }
          }
        >
          Manage your guardian information
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 24,
            },
            {
              "flex": 1,
            },
            {
              "fontFamily": "feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 30,
            },
            {
              "flex": 1,
            },
            {
              "fontFamily": "anticon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          {
            "flex": 6,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Medication Database
        </Text>
        <Text
          style={
            {
              "fontSize": 12,
            }
          }
        >
          Search through our database for your medication
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 24,
            },
            {
              "flex": 1,
            },
            {
              "fontFamily": "feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 30,
            },
            {
              "flex": 1,
            },
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
      <View
        style={
          {
            "flex": 6,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Log Out
        </Text>
        <Text
          style={
            {
              "fontSize": 12,
            }
          }
        >
          Log out of your existing account
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 24,
            },
            {
              "flex": 1,
            },
            {
              "fontFamily": "feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flex": 3,
        "flexDirection": "column",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        {
          "flex": 1,
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      More
    </Text>
    <View
      style={
        {
          "borderColor": "black",
          "borderRadius": 15,
          "borderWidth": 1,
          "flex": 4,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 24,
              },
              {
                "flex": 1,
              },
              {
                "fontFamily": "material",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            {
              "flex": 6,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            Help & Support
          </Text>
        </View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 24,
              },
              {
                "flex": 1,
              },
              {
                "fontFamily": "feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 24,
              },
              {
                "flex": 1,
              },
              {
                "fontFamily": "ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
        <View
          style={
            {
              "flex": 6,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            About App
          </Text>
        </View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "black",
                "fontSize": 24,
              },
              {
                "flex": 1,
              },
              {
                "fontFamily": "feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
